Feature/generate qrcode for add contact

kurotyan514 opened the pull request:
* STORY
https://github.com/ReadyResponder/ReadyResponder/issues/2

add QRCode on person show page, user can use their phone to scan and have them added to their contacts.

please review by commit thanks! and feel free to ask

<img width="540" alt="2017-10-20 5 03 40" src="https://user-images.githubusercontent.com/14138733/31813380-b7898fb8-b54b-11e7-97ed-70fb21c7c61f.png">

kurotyan514 committed:
add gems

- `rqrcode` for generate QRcode
- `vcardigan` for generate vcard info

kurotyan514 committed:
modify CATEGORIES for vcard info

kurotyan514 committed:
add services for generate QRCode

- add `QrCode::Generator` for generate QRCode
- add `Vcard::Generator` for generate vcard

kurotyan514 committed:
render QRCode on person show page

- add QRCode css
- render QRCode on show action
- show the QRCode on sidebar

kurotyan514 committed:
refactor

Extract vcard from QRCode , make QRCode can be more reuseable

kgf commented on app/services/vcard/generator.rb:
```diff
@@ -0,0 +1,37 @@
+    end
+
+    def gender_abbr
+      person.gender == 'Male' ? 'Mr.' : 'Mrs.'
```
I think we probably don't need the gender. In America, `Mrs` signifies a married woman only.

kgf commented on app/services/vcard/generator.rb:
```diff
@@ -0,0 +1,37 @@
+      vcard.fullname person.fullname
+      vcard.email person.email
+      vcard.bday person.date_of_birth.strftime("%Y%m%d")
+      vcard.title person.department.name
```
Could we do this as the `org` instead of the title?

kgf commented on app/views/people/show.html.erb:
```diff
@@ -4,6 +4,7 @@
   <%= sidebar_button_link 'New Timecard', new_person_timecard_path(@person) if can? :create, Timecard %>
   <br>
+  <%= raw @qr %>
   <%= image_tag @person.portrait_url if @person.portrait.file.present? %>
```
Please put the portrait above the barcode. Thanks

kgf commented:
This is very good! Some small changes requested.

kurotyan514 committed:
revise code review

- remove fender info from vcard
- use org insteds of title
- change barcode postisionn

pcai commented on app/services/vcard/generator.rb:
```diff
@@ -0,0 +1,33 @@
+    end
+
+    def call
+      @vcard = VCardigan.create(version: VERSION)
```
is this a typo? its declared `@vcard` here but referenced as `vcard` in the rest of the method

pcai replied:
sorry, I just saw the `attr_reader :vcard` declaration. But I am not so sure about the value of an instance variable that is only ever used in the same invocation of the method that sets it?

kgf committed:
Merge branch 'development' into feature/generate_qrcode_for_add_contact

kgf commented:
@kurotyan514 In order to fix the test suite, you need to add `date_of_birth 25.years.ago` to the file at `spec/factories/person.rb`
Thanks for your help,
Kevin
